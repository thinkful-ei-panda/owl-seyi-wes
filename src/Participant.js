import React from 'react';

function Participant(props){
    if (props.participant.inSession===true&&props.participant.onStage===false){
        return  (<div className='participant-container'>
                    <div className='left small'>
                        <img src={props.participant.avatar} alt='avatar' />
                    </div>
                    <div className='right'>
                        <div>{props.participant.name}</div>
                        <div>in session</div>
                    </div>
                </div>)
    } else if(props.participant.inSession===true&&props.participant.onStage===true){
        return  (<div className='participant-container'>
                    <div className='left small'>
                        <img src={props.participant.avatar} alt='avatar' />
                    </div>
                    <div className='right'>
                        <div>{props.participant.name}</div>
                        <div>on stage</div>
                    </div>
                </div>)
    } else{
        return <></>
    }
}

export default Participant