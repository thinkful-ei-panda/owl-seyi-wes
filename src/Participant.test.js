import React from 'react';
import ReactDOM from 'react-dom';
import Participant from './Participant';

import renderer from 'react-test-renderer';


describe('Participant', () => {

    const props={
        key:0,
        participant:{
            id: 1,
            name: 'Koren Templeton',
            avatar:
                'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
            inSession: true,
            onStage: true
        }
    }
  it('renders without crashing', () => {

    const div = document.createElement('div');
    ReactDOM.render(<Participant {...props}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Participant {...props}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

});

