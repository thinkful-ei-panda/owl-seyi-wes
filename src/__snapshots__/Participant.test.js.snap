// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant renders the UI as expected 1`] = `
<div
  className="participant-container"
>
  <div
    className="left small"
  >
    <img
      alt="avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="right"
  >
    <div>
      Koren Templeton
    </div>
    <div>
      on stage
    </div>
  </div>
</div>
`;
